geps/gep-1767/index.md

```diff
@@ -0,0 +1,701 @@
+    // Support: Extended
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=64
+    AllowOrigins []string `json:"allowOrigins,omitempty"`
+ 
+    // AllowCredentials indicates whether the actual cross-origin request 
+    // allows to include credentials.
+    //
+    // The only valid value for the header `Access-Control-Allow-Credentials` is true (case-sensitive).
+    //
+    // The default value of the config AllowCredentials is false.
+    // If the credentials are not allowed in cross-origin requests, 
+    // the gateway will omit the header `Access-Control-Allow-Credentials` entirely.
+    //
+    // Input:
+    //   Origin: https://foo.example
+    //
+    // Config:
+    //   allowCredentials: true
+    //
+    // Output:
+    //   Access-Control-Allow-Origin: https://foo.example
+    //   Access-Control-Allow-Credentials: true
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:default=false
+    AllowCredentials bool `json:"allowCredentials,omitempty"`
+
+    // AllowMethods indicates which HTTP methods are supported 
+    // for accessing the requested resource.
+    // The method is case-sensitive.
+    //
+    // Config:
+    //   allowMethods: ["GET, PUT, POST, DELETE, PATCH, OPTIONS"]
+    // 
+    // Output:
+    //   Access-Control-Allow-Methods: GET, PUT, POST, DELETE, PATCH, OPTIONS
+    //
+    // A wildcard indicates that the requests with all HTTP methods are allowed.
+    // When responding to a credentialed requests, the gateway must specify 
+    // one or more HTTP methods in the value of the Access-Control-Allow-Methods response header, 
+    // instead of specifying the * wildcard.
+    //
+    // Config:
+    //   allowMethods: ["*"]
+    // 
+    // Output:
+    //   Access-Control-Allow-Methods: ["*"]
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=16
+    AllowMethods []string `json:"allowMethods,omitempty"`
+ 
+    // AllowHeaders indicates which HTTP request headers are supported 
+    // for accessing the requested resource.
+    // Header names are not case sensitive.
+    //
+    // Config:
+    //   allowHeaders: ["DNT", "Keep-Alive", "User-Agent", "X-Requested-With", "If-Modified-Since", "Cache-Control", "Content-Type", "Range", "Authorization"]
+    // Output:
+    //   Access-Control-Allow-Headers: DNT, Keep-Alive, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, Authorization
+    //
+    // A wildcard indicates that the requests with all HTTP headers are allowed.
+    // When responding to a credentialed requests, the gateway must specify 
+    // one or more HTTP headers in the value of the Access-Control-Allow-Headers response header, 
+    // instead of specifying the * wildcard.
+    //
+    // Config:
+    //   allowHeaders: ["*"]
+    // Output:
+    //   Access-Control-Allow-Headers: *
+    //
+    // Support: Extended
+    //
+    // +optional
```

Comment:
As MDN mentions,

> This header is required if the preflight request contains [Access-Control-Request-Headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Request-Headers)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers


I would recommend echoing back the `Access-Control-Request-Headers` request header by default, so the user doesn't need to study what a credentialed request means.